compiler/damlc/tests/daml-test-files/Swappable.daml

```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE UndecidableSuperClasses #-}
```

Comment:
I don’t think we want to encourage people to use `UndecidableSuperClasses`. @tweber-da had a legitimate usecase that required this yesterday and got confused by the error so if our expert DAML users get confused by this, it is not acceptable to expose people to this for cases where it is only required due to preprocessing so the error doesn’t even make sense.

We could turn on `UndecidableSuperClasses` by default but I’m not at all convinced that is a good idea.

Reply:
Following our discussion, we've decided to turn `UndecidableSuperClasses` on by default. If there's an actual loop, GHC will abort after a limited number of iterations without going into an infinite loop.